torch/testing/_internal/distributed/_sharded_tensor/__init__.py

```diff
@@ -15,13 +15,30 @@ class ShardedTensorTestBase(MultiProcessTestCase):
     def world_size(self):
         return 4
 
-    def init_pg(self):
-        dist.init_process_group(
-            backend="nccl",
-            world_size=self.world_size,
-            rank=self.rank,
-            init_method=f"file://{self.file_name}",
-        )
+    def init_pg(self, backend="nccl"):
+        if backend == "nccl":
```

Comment:
probably just 

if backend not in ["nccl", "gloo", "mpi"]  --> raise error 

and then 

dist.init_process_group(
                backend=backend,
                world_size=self.world_size,
                rank=self.rank,
                init_method=f"file://{self.file_name}",
            )